it/test/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepositoryItSpec.scala

```diff
@@ -288,15 +296,91 @@ class MovementRepositoryItSpec extends PlaySpec
         val result = repository.getMovementByERN(Seq("ern1")).futureValue
 
         result.sortBy(_.localReferenceNumber) mustBe Seq(
-          expectedMovement1, expectedMovement2, expectedMovement3
+          expectedMovement1,
+          expectedMovement2,
+          expectedMovement3
+        )
+      }
+      "lrn from filter matches record from database" in {
+        val expectedMovement1 =
+          Movement(Some("boxId"), "1", "consignorId1", Some("ern1"), Some("arc1"), lastUpdated = timestamp)
+        val expectedMovement2 =
+          Movement(Some("boxId"), "2", "ern1", Some("ern2"), Some("arc2"), lastUpdated = timestamp)
+        val expectedMovement3 =
+          Movement(Some("boxId"), "3", "consignorId1", Some("ern1"), Some("arc3"), lastUpdated = timestamp)
+        val expectedMovement4 = Movement(Some("boxId"), "4", "ern4", None, Some("arc4"), lastUpdated = timestamp)
+        insertMovement(expectedMovement1)
+        insertMovement(expectedMovement2)
+        insertMovement(expectedMovement3)
+        insertMovement(expectedMovement4)
+
+        val result =
+          repository.getMovementByERN(Seq("ern1"), MovementFilter.emptyFilter.copy(lrn = Some("1"))).futureValue
+
+        result mustBe Seq(
+          expectedMovement1
+        )
+
+      }
+
+      "arc from filter matches record from database" in {
+        val expectedMovement1 =
+          Movement(Some("boxId"), "1", "consignorId1", Some("ern1"), Some("arc1"), lastUpdated = timestamp)
+        val expectedMovement2 =
+          Movement(Some("boxId"), "2", "ern1", Some("ern2"), Some("arc2"), lastUpdated = timestamp)
+        val expectedMovement3 =
+          Movement(Some("boxId"), "3", "consignorId1", Some("ern1"), Some("arc3"), lastUpdated = timestamp)
+        val expectedMovement4 = Movement(Some("boxId"), "4", "ern4", None, Some("arc4"), lastUpdated = timestamp)
+        insertMovement(expectedMovement1)
+        insertMovement(expectedMovement2)
+        insertMovement(expectedMovement3)
+        insertMovement(expectedMovement4)
+
+        val result =
+          repository.getMovementByERN(Seq("ern1"), MovementFilter.emptyFilter.copy(arc = Some("arc1"))).futureValue
+
+        result mustBe Seq(
+          expectedMovement1
         )
+
+      }
+
+      "updatedSince from filter matches record from database" in {
```

Comment:
This test is only really testing that movements come back for "ern1" the lastUpdated isn't tested here as they are all set as the same timestamp and the filter timestamp will allow all the movements for "ern1" to be returned. Can we have tests for the updatedSince like the deleted tests in the MovementFilter i.e. test only ones after the filter time stamp come back, test only ones with the same filter timestamp come back and test ones where the timestamp is before the updatedSince that they don't come back? You can do things like `instant.minus(5, ChronoUnit.MINUTES)` to get timestamps 5 minutes in the past or whatever you need, which might be better for testing, instead of fixed timestamps?

Reply:
Done :)